### PR TITLE
fix: inconsistent evaluation of tm_try()

### PR DESCRIPTION
**Pull request by katcipis**

### CODE REVIEW

File: stack/globals_test.go [hunk 1]
```diff
@@ -653,7 +653,28 @@ func TestLoadGlobals(t *testing.T) {
...
+						expr("b", `tm_try(global.undefined, global.a)`),
+						expr("c", `tm_try(global.a, global.undefined)`),
+					),
+				},
+			},
+			want: map[string]*hclwrite.Block{
+				"/stack": globals(
+					str("a", "value"),
+					str("b", "value"),
+					str("c", "value"),
```
Comment (mariux) [inline stack/globals_test.go:672]: :heart:

File: stack/globals_test.go [hunk 2]
```diff
@@ -830,6 +851,19 @@ func TestLoadGlobals(t *testing.T) {
...
 			wantErr: errors.E(stack.ErrGlobalEval),
 		},
+		{
+			name:   "global interpolating undefined reference fails",
+			layout: []string{"s:stack"},
+			configs: []hclconfig{
+				{
+					path: "/stack",
+					add: globals(
+						str("a_interpolated", "${global.undefined}-something"),
```
Comment (mariux) [inline stack/globals_test.go:861]: :+1:

**Review by mariux — APPROVED** [review]
lgtm

this should not be a breaking change afaiu?

it's a change in behavior but stuff that worked before is still the same result?

**Review by i4ki — APPROVED** [review]
:rocket:

**Comment by katcipis** [comment]
> lgtm
> 
> this should not be a breaking change afaiu?
> 
> it's a change in behavior but stuff that worked before is still the same result?

It should not be a breaking change, the only change in behavior is that code that used to fail before will now work, if you are pedantic it could be considered a breaking change, because maybe some people depended on the breaking/failing behavior, but I don't think we need to be that pedantic at this point =P

**Pull request merged**